src/containers/consignment.rs

```diff
@@ -376,6 +376,31 @@ impl<const TRANSFER: bool> Consignment<TRANSFER> {
                 )));
             }
         }
+        index.bundle_ids().for_each(|bundle_id| {
+            let transition_bundle = index
+                .bundle(bundle_id)
+                .expect("index transition bundle failed");
+            transition_bundle
+                .known_transitions
+                .values()
+                .for_each(|transition| {
+                    transition.assignments.values().for_each(|assign| {
+                        assign.as_attachment().iter().for_each(|item| {
+                            let state = item
+                                .as_revealed_state()
+                                .expect("get revealed attached state failed");
+                            if !self.attachments.keys().any(|&id| id == state.id) {
```

Comment:
```suggestion
                            if !self.contains(state.id) {
```

Reply:
There is a question: it's not clear to me why `self.contains`, because `self` is for the consignment, and no `contains` method here.

Reply:
Sorry it should be `self.attachments.contains_key`